src/main/kotlin/com/exactpro/th2/dataprocessor/zephyr/impl/cache/LRUCache.kt

```diff
@@ -16,27 +16,70 @@
 
 package com.exactpro.th2.dataprocessor.zephyr.impl.cache
 
+import mu.KotlinLogging
 import org.apache.commons.collections4.map.LRUMap
-import java.util.function.LongSupplier
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+import java.time.LocalDate
+import java.time.LocalTime
+import java.time.ZoneOffset
+import java.time.temporal.ChronoUnit
 
 internal class LRUCache<K, V>(
     size: Int,
     private val expireAfterMillis: Long,
-    private val timeSource: LongSupplier = LongSupplier(System::currentTimeMillis),
+    private val timeSource: Clock = Clock.systemUTC(),
+    invalidateAt: LocalTime? = null,
 ) {
+    private var nextInvalidateAt: Instant
```

Comment:
Should this variable have volatile modifier?

Reply:
Probably, yes. Good catch!